### PR TITLE
Improve course point distance/altitude

### PR DESCRIPTION
**Pull request by Ptosiek**
compute course_point distance/altitude from tcx file directly if possible
fix (partly) #44
add test (to be improved)

So that's how it could look like. Now for the file used in test here are the differences:

**Using tcx data:**
_distance_
```
[ 0.        0.017529  0.027187  0.407902  0.645253  5.529111  6.015854
  6.385713  6.562241  7.12056   7.135924  8.813289  9.6664   10.254984
 10.394647 10.407042 11.415968 12.286501]
```

_altitude_
```
[45.8 45.7 45.6 48.5 49.2 36.3 32.3 28.7 27.  31.6 31.9 37.8 30.  41.
 39.6 39.4 31.1 45.5]
 ```
 
 **doing the calculation:**
 _distance_
 ```
 [ 0.          0.01648527  0.02495816  0.40788932  0.59441527  5.52911168
  6.01581745  6.38571265  6.56223896  7.12055985  7.13592386  8.81306384
  9.66379496 10.25417516 10.39443439 10.39998769 11.40894897 12.28649897]
  ```
  
  _altitude_
  ````
  [46.15790536 46.17606598 46.18539994 46.04656585 46.09408459 39.8528199
 40.0873497  40.34487467 40.83065774 41.30839913 41.67492115 42.35886119
 43.60414905 44.03182548 44.08607776 44.08821497 44.47651762 45.0031213 ]
```

Not sure what to think. There might well be errors in the tcx file in the first place and we might be passing them on, but the errors might be small enough (distance are close, altitude not that much) that it's easier than implementing the proper logic to calculate the distance/altitude back after downsampling. What do you think ?

### CODE REVIEW

**Comment by hishizuka** [comment]
Hi, @Ptosiek ,
let's see on a course that has a lot of climbing.
I am also wondering how much slower the process will be on the Raspberry Pi Zero. I will experiment.

**Comment by Ptosiek** [comment]
On a side note:
            end_distance = self.config.get_dist_on_earth_array(
                self.longitude[-1],
                self.latitude[-1],
                course_points.longitude[-1],
                course_points.latitude[-1],
            )
there's no need to use get_dist_on_earth_array (normal get_dist_on_earth) here, or am I missing something ?

File: modules/course.py
Comment (hishizuka) [inline modules/course.py:1]: Sorry, let setxattr/getxattr and shutil.copy2 be turned off temporarily.
It does not work in our environment. (macOS and UserLAnd on Android)

**Review by hishizuka — APPROVED** [review]

**Comment by hishizuka** [comment]
Hi @Ptosiek ,
thank you for your PR.

> Not sure what to think. There might well be errors in the tcx file in the first place and we might be passing them on, but the errors might be small enough (distance are close, altitude not that much) that it's easier than implementing the proper logic to calculate the distance/altitude back after downsampling. What do you think ?

I have tried it on other courses and it is almost right.
The distance is off by 1-2 meters in most cases, sometimes by several tens of meters. This is probably a bug in my logic. I don't care too much.
Elevation is smoothed by savitzky_golay, so it will be shifted. This elevation is not used elsewhere and is not affected. I used to plot with `course_profile`, but as you can see, the climb segment is more useful information, so I deleted it.

The impact on execution speed is also acceptable. Thanks.
The execution time of read_file increased by 10% and modify_course_points has zero execution time. It was only a small increase of about 7 seconds total for a course with 15000 coordinates and 150 course points with a distance of 300 km.

File: modules/course.py
Comment (Ptosiek) [reply modules/course.py:1]: No problem

**Pull request merged**